File: Lua/GameFKBY/View/SafeValidate.lua
local this = LuaObject:New()
SafeValidate = this;

this.ValidateType = {BankPassword = 0, PhoneCode = 1}

local mCheckState = false;
local mLoginState = false;
local mValidateType;

local cjson = require "cjson"

function this.DoCheckValidateType(  )
	if mCheckState == false then
		UIHelper.ShowProgressHUD(nil,"");
		local www = HttpConnect.Instance:HttpRequestWithSession(ConnectDefine.SAFE_VALIDATE_TYPE_URL, nil);
		coroutine.www(www);
		print(www.text);
		UIHelper.HideProgressHUD();
		local js = cjson.decode(www.text);
		if js["result"] == "ok" then
			mCheckState = true;
			mValidateType = js["body"]["bank_validate"];
			local temp = js["body"]["is_login"];
			local isLogin;
			if temp == 1 then
				isLogin = true;
			else
				isLogin = false;
			end
			this.UpdateLoginState(isLogin);
		else
			this.UpdateLoginState(false);
			UIHelper.ShowMessage(js["body"],2);
		end
	end
end
function this.OnClickSendPhoneCode(  )
	UIHelper.ShowProgressHUD();
	local www = HttpConnect.Instance:HttpRequestWithSession(ConnectDefine.SAFE_SEND_PHONECODE_URL, nil);
	coroutine.www(www);
	print(www.text);
	UIHelper.HideProgressHUD();
	local js = cjson.decode(www.text);
	if js["result"] == "ok" then 
		UIHelper.ShowMessage("HttpConnectSucess",2);
	else
		UIHelper.ShowMessage(js["body"],2);
	end
end
function this.OnClickValidate( password , func)
	if mLoginState == false then
		UIHelper.ShowProgressHUD(nil,"");
		local form = UnityEngine.WWWForm.New();
		form:AddField("password",password);
		local  www = HttpConnect.Instance:HttpRequestWithSession(ConnectDefine.SAFE_VALIDATE_LOGIN_URL,form);
		coroutine.www(www);
		print(www.text);
		UIHelper.HideProgressHUD();
		local js = cjson.decode(www.text);
		if js["result"] == "ok" then
			this.UpdateLoginState(true);
		else
			this.UpdateLoginState(false);
			UIHelper.ShowMessage(js["body"],2);
		end
	end
	if func ~= nil then func() end;
end
function this.UpdateLoginState(loginState)
	mLoginState = loginState;
	if loginState == true then
		this.HideValidateInput();
	else
		this.ShowValidateInput(mValidateType);
	end

end
function this.LoginState()
	return mLoginState;
end
function this.GetValidateType( )
	return mValidateType;
end

function this.ShowValidateInput( validateType )
	-- body
end
function this.HideValidateInput(  )
	
end

File: __HappyCity/Scripts/Scene/Module/SafeValidate.cs
using UnityEngine;
using System.Collections;

public class SafeValidate : ModuleBase {

	public enum ValidateType {
		BankPassword = 0, 
		PhoneCode = 1,
	}
	
	private bool mCheckState = false;
	private bool mLoginState = false;
	private ValidateType mValidateType;
	
	/* ------ Button Click ------ */
	public IEnumerator DoCheckValidateType () {
		if (!mCheckState) {
			EginProgressHUD.Instance.ShowWaitHUD(ZPLocalization.Instance.Get("HttpConnectWait"));
			
			WWW www = HttpConnect.Instance.HttpRequestWithSession(ConnectDefine.SAFE_VALIDATE_TYPE_URL, null);
			yield return www;
			
			HttpResult result = HttpConnect.Instance.BaseResult(www);		
			EginProgressHUD.Instance.HideHUD();
			if(HttpResult.ResultType.Sucess == result.resultType) {
				mCheckState = true;
				JSONObject resultObj = (JSONObject)result.resultObject;
				mValidateType = (ValidateType)resultObj["bank_validate"].n;
				bool isLogin = (resultObj["is_login"].n == 1);
				UpdateLoginState(isLogin);
			}else {
				UpdateLoginState(false);
				EginProgressHUD.Instance.ShowPromptHUD(result.resultObject as string);
			}
		}
	}

	IEnumerator OnClickSendPhoneCode () {
		EginProgressHUD.Instance.ShowWaitHUD(ZPLocalization.Instance.Get("HttpConnectWait"));
		
		WWW www = HttpConnect.Instance.HttpRequestWithSession(ConnectDefine.SAFE_SEND_PHONECODE_URL, null);
		yield return www;
		
		HttpResult result = HttpConnect.Instance.BaseResult(www);	
		EginProgressHUD.Instance.HideHUD();
		if(HttpResult.ResultType.Sucess == result.resultType) {
			EginProgressHUD.Instance.ShowPromptHUD(ZPLocalization.Instance.Get("HttpConnectSucess"));
		}else {
			EginProgressHUD.Instance.ShowPromptHUD(result.resultObject as string);
		}
	}
	
	public IEnumerator OnClickValidate (string password) {
		if (!mLoginState) {
			EginProgressHUD.Instance.ShowWaitHUD(ZPLocalization.Instance.Get("HttpConnectWait"));
			WWWForm form = new WWWForm();
			form.AddField("password", password);
			WWW www = HttpConnect.Instance.HttpRequestWithSession(ConnectDefine.SAFE_VALIDATE_LOGIN_URL, form);
			yield return www;
			
			HttpResult result = HttpConnect.Instance.BaseResult(www);	
			EginProgressHUD.Instance.HideHUD();
			if(HttpResult.ResultType.Sucess == result.resultType) {
				UpdateLoginState(true);
			}else {
				UpdateLoginState(false);
				EginProgressHUD.Instance.ShowPromptHUD(result.resultObject as string);
			}
		}
	}
	
	/* ------ Other ------ */
	private void UpdateLoginState (bool loginState) {		
		mLoginState = loginState;
		if (loginState) {
			HideValidateInput();
		}else {
			ShowValidateInput(mValidateType);
		}
	}
	
	public bool LoginState {
		get {
			return mLoginState;
		}
	}
	
	public ValidateType GetValidateType {
		get {
			return mValidateType;
		}
	}
	
	/* ------ Virtual ------ */
	public virtual void ShowValidateInput (ValidateType validateType) {}
	
	public virtual void HideValidateInput () {}
}
